AirCasting/Graph/Graph.swift

```diff
@@ -214,6 +225,48 @@ struct Graph: UIViewRepresentable {
             return chartDataEntry
         }) ?? []
         uiView.updateWithEntries(entries: entries, isAutozoomEnabled: isAutozoomEnabled)
+        
+        let limitLines = getLimitLines()
+        if limitLines != midnightLines {
+            midnightLines = limitLines
+            uiView.updateMidnightLines(with: limitLines)
+        }
+    }
+    
+    func getMidnightsPoints(startingDate: Date) -> [Double] {
+        let now = Date()
+        var midnight = Calendar.current.date(bySettingHour: 0, minute: 0, second: 0, of: now)!
+        let components = Calendar.current.dateComponents([.hour, .minute, .second], from: midnight)
+        let firstMeasurementDate = startingDate
+        var midnightPoints: [Double] = []
+        
+        while (midnight > firstMeasurementDate) {
+            let previus = Calendar.current.nextDate(after: midnight,
+                                                    matching: components,
+                                                    matchingPolicy: Calendar.MatchingPolicy.previousTimePreservingSmallerComponents,
+                                                    repeatedTimePolicy: Calendar.RepeatedTimePolicy.first,
+                                                    direction: .backward)
+            if let midnightPoint = previus?.timeIntervalSince1970 {
+                midnightPoints.append(midnightPoint)
+            }
+            midnight = previus!
+        }
+        return midnightPoints
+    }
+    
+    func getLimitLines() -> [ChartLimitLine] {
+        let points = getMidnightsPoints(startingDate: stream.allMeasurements?.first?.time ?? Date())
+        var limitLines: [ChartLimitLine] = []
+        
+        for point in points {
```

Comment:
It's generally considered a good practice in Swift to use [higher order functions](https://engineering.nodesagency.com/categories/ios/2019/07/03/Higher-Order-Functions-In-Swift) instead of classic `for/while/do`. In this example I recommend using the [map(:_)](https://developer.apple.com/documentation/swift/array/3017522-map):
``` Swift
return points.map {
   let line = ChartLimitLine(limit: $0)
   line.lineColor = UIColor.aircastingDarkGrey
   ...
}
```